File: bindings/imgui_bundle/demos_cpp/demo_implot.cpp
// Part of ImGui Bundle - MIT License - Copyright (c) 2022-2026 Pascal Thomet - https://github.com/pthom/imgui_bundle
#if defined(IMGUI_BUNDLE_WITH_IMPLOT) && defined(IMGUI_BUNDLE_WITH_IMPLOT3D)

#include "implot/implot.h"
#include "implot3d/implot3d.h"
#include "immapp/immapp.h"
#include "hello_imgui/icons_font_awesome_4.h"
#ifdef IMGUI_BUNDLE_WITH_IMGUI_EXPLORER_LIB
#include "imgui_explorer.h"
#endif


void demo_implot()
{
    ImGuiMd::RenderUnindented(R"(
        [Implot](https://github.com/epezent/implot) and [Implot3D](https://github.com/brenocq/implot3d) are fast and efficient libraries which provide immediate Mode Plotting.
    )");
    if (ImGui::CollapsingHeader("ImPlot: Full Demo"))
    {
        ImGui::Text(ICON_FA_INFO_CIRCLE);
        if (ImGui::IsItemHovered())
            ImGuiMd::RenderUnindented(R"(
                > [!TIP]
                > Python users: `implot_ctx` provides context managers for the begin/end and push/pop pairs (see "Demo Apps" tab, demo_python_context_manager).
                )");

#ifdef IMGUI_BUNDLE_WITH_IMGUI_EXPLORER_LIB
        ImGui::PushID("ImPlotDemo");
        ShowImGuiExplorerGui_Cpp(ImGuiExplorerLibrary::ImPlot);
        ImGui::PopID();
#else
        ImPlot::ShowDemoWindow_MaybeDocked(false);
#endif
    }

    if (ImGui::CollapsingHeader("ImPlot3D: Full Demo"))
    {
#ifdef IMGUI_BUNDLE_WITH_IMGUI_EXPLORER_LIB
        ImGui::PushID("ImPlot3DDemo");
        ShowImGuiExplorerGui_Cpp(ImGuiExplorerLibrary::ImPlot3D);
        ImGui::PopID();
#else
        ImPlot3D::ShowAllDemos();
#endif
    }
}

#else // defined(IMGUI_BUNDLE_WITH_IMPLOT) && defined(IMGUI_BUNDLE_WITH_IMPLOT3D)
#include "imgui.h"
void demo_implot() { ImGui::Text("Dear ImGui Bundle was compiled without support for both ImPlot and ImPlot3D"); }
#endif

File: bindings/imgui_bundle/demos_python/demo_implot.py
# Part of ImGui Bundle - MIT License - Copyright (c) 2022-2026 Pascal Thomet - https://github.com/pthom/imgui_bundle
from imgui_bundle import imgui, imgui_md, immapp
from imgui_bundle.demos_python.demos_imgui_explorer import implot3d_demo
from imgui_bundle.demos_python.demos_imgui_explorer import implot_demo

from imgui_bundle.demos_python.demo_utils.imgui_explorer_setup import get_imgui_explorer, get_package_path
imgui_explorer, _has_imgui_explorer = get_imgui_explorer()


def demo_gui():
    imgui_md.render_unindented(
        """
        [Implot](https://github.com/epezent/implot) and [Implot3D](https://github.com/brenocq/implot3d) are fast and efficient libraries which provide immediate Mode Plotting.
        """
    )

    if imgui.collapsing_header("ImPlot: Full Demo"):
        if _has_imgui_explorer:
            imgui.push_id("ImPlotDemo")
            imgui_explorer.show_imgui_explorer_gui_python(imgui_explorer.ImGuiExplorerLibrary.implot, get_package_path())
            imgui.pop_id()
        else:
            implot_demo.show_all_demos()

    if imgui.collapsing_header("ImPlot3D: Full Demo"):
        if _has_imgui_explorer:
            imgui.push_id("ImPlot3DDemo")
            imgui_explorer.show_imgui_explorer_gui_python(imgui_explorer.ImGuiExplorerLibrary.implot3_d, get_package_path())
            imgui.pop_id()
        else:
            implot3d_demo.show_all_demos()


def main():
    immapp.run(demo_gui, with_implot=True, with_implot3d=True, with_markdown=True, window_size=(1000, 800))


if __name__ == "__main__":
    main()
